File: dotnet/SDL.DXA.Modules.FlexLayout/Models/TitledEntityContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SDL.DXA.Modules.FlexLayout.Models
{
    /// <summary>
    /// Titled Content Container
    /// General representation of a layout container with a navigation bar.The navigation bar has a title coming from entity.
    /// Typical examples are tabs and accordions.
    /// </summary>
    public class TitledEntityContainer : LayoutContainer
    {
        private List<TitledEntity> titledEntities = null;

        /// <summary>
        /// Get list of titled entities
        /// </summary>
        public IList<TitledEntity> TitledEntities
        {
            get
            {
                if ( titledEntities == null )
                {
                    titledEntities = new List<TitledEntity>();
                    foreach ( var entity in Region.Entities )
                    {
                        titledEntities.Add(new TitledEntity(entity));
                    }
                }
                return titledEntities;
            }
        }
    }
}

File: java/dxa-module-flex-layout/src/main/java/com/sdl/dxa/modules/flexlayout/model/TitledEntityContainer.java
package com.sdl.dxa.modules.flexlayout.model;

import com.sdl.webapp.common.api.model.EntityModel;
import com.sdl.webapp.common.exceptions.DxaException;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * Titled Content Container
 * General representation of a layout container with a navigation bar. The navigation bar has a title coming from entity.
 * Typical examples are tabs and accordions.
 *
 * @author nic
 */
public class TitledEntityContainer extends LayoutContainer {

    private List<TitledEntity> titledEntities = null;

    public TitledEntityContainer() throws DxaException {

    }

    /**
     * Get list of titled entities
     * @return list
     * @throws DxaException
     */
    public Collection<TitledEntity> getTitledEntities() throws DxaException {
        if ( titledEntities == null ) {
            titledEntities = new ArrayList<>();
            for ( EntityModel entity : this.getRegion().getEntities() ) {
                titledEntities.add(new TitledEntity(entity));
            }
        }
        return titledEntities;
    }
}
